diff --git a/tldextract/tldextract.py b/tldextract/tldextract.py
index ea20f02..361fc33 100644
--- a/tldextract/tldextract.py
+++ b/tldextract/tldextract.py
@@ -257,6 +257,8 @@ class TLDExtract:
     def _extract_netloc(
         self, netloc: str, include_psl_private_domains: Optional[bool]
     ) -> ExtractResult:
+        if include_psl_private_domains is None:
+            include_psl_private_domains = self.include_psl_private_domains
         labels = (
             netloc.replace("\u3002", "\u002e")
             .replace("\uff0e", "\u002e")
